app/controllers/editions_controller.rb

```diff
@@ -440,4 +442,29 @@ def progress_action_param
   def attempted_activity
     Edition::ACTIONS.invert[params[:commit]]
   end
+
+  def construct_objects_with_errors
+    @objects_with_errors = []
+
+    @objects_with_errors << @resource if @resource.errors.present?
+
+    return unless @resource.instance_of?(SimpleSmartAnswerEdition)
+
+    if @resource.nodes.present?
+      @resource.nodes.each do |node|
+        @objects_with_errors << node if node.errors.present?
+        next if node.options.blank?
+
+        node.options.each do |option|
+          @objects_with_errors << option if option.errors.present?
+        end
+      end
+    end
+
+    @objects_with_errors.each do |object|
+      object.errors.errors.reject! do |error|
+        error.type == :invalid
+      end
+    end
+  end
```

Comment:
Yea good shout. As discussed i think this should probs live at the model level like it does for guide so it can be unit tested. Guides need to be added to this anyway so something like 

```
def construct_object_with_errors
  case @resource
  when @resource.class == SimpleSmartAnswer || @resource.class == Guide
    @resource.construct_errors_from_self_and_nested_objects
  else 
    @resource.errors
  end
end
```

seems more appropriate, then let the model handle the rest so we can unit test it.